homeassistant/helpers/entity_registry.py

```diff
@@ -1224,6 +1237,15 @@ def _data_to_save(self) -> dict[str, Any]:
 
         return data
 
+    @callback
+    def async_clear_category_id(self, scope: str, category_id: str) -> None:
+        """Clear category id from registry entries."""
+        for entity_id, entry in self.entities.items():
+            if scope in entry.categories and category_id == entry.categories[scope]:
```

Comment:
A `entry.categories.get(scope)` should be faster as it only needs to lookup the `scope` key once in the dictionary.

Reply:
Addressed in [deb45eb](https://github.com/home-assistant/core/pull/110897/commits/deb45eb3cbb1137b50322a63073bcbb6eb20e44a)